### PR TITLE
[feat] moe lora

### PR DESCRIPTION
**Pull request by Jackmin801**
I was too lazy to explain my changes, decreasing my chances of this ever getting merged.

---

**GitHub Issue**: [Issue ID]
**Linear Issue**: Resolves [Issue ID]

---

> [!NOTE]
> Enables training LoRA adapters on MoE experts and wires support across trainer, packing, metrics, and exports.
> 
> - Introduces `MultiLoRAGroupedExperts` that applies LoRA to MoE `w1/w2/w3`, supports Hopper `grouped_mm` with fallback, exposes `named_parameters_for_adapter`, `get_lora_param_counts`, and vLLM-compatible `state_dict_for_adapter`
> - `apply_lora_to_model` now targets both `nn.Linear` and `GroupedExperts`; default `lora.target_modules` adds `"experts"`; `set_multilora_offsets` extended to MoE implementation
> - Perf and logging updated to use `MultiLoRAModule.get_lora_param_counts()` instead of direct `lora_A/B` access
> - Packer repacks microbatches per `run_idx` (no mixing across runs) to satisfy current MultiLoRA MoE constraints
> - `Runs.get_state_dict_for_run` emits per-expert adapter tensors when modules provide `state_dict_for_adapter`
> - Exports updated in `lora/__init__.py`; CHANGELOG notes QwenMoE expert LoRA support by default
> 
> <sup>Written by [Cursor Bugbot](https://cursor.com/dashboard?tab=bugbot) for commit 8876caf60a572d72efd3b2ce432dfe329367ca7c. This will update automatically on new commits. Configure [here](https://cursor.com/dashboard?tab=bugbot).</sup>

### CODE REVIEW

**Comment by Jackmin801** [comment]
@codex review

**Comment by Jackmin801** [comment]
@cursoragent add dropout the for loop path of MultiLoRAMoE

File: src/prime_rl/trainer/models/layers/lora/multi_moe.py [hunk 1]
```diff
@@ -0,0 +1,433 @@
...
+
+        return torch.cat(out_splits, dim=0)
+
+    def __repr__(self) -> str:
+        return (
+            f"{self.__class__.__name__}(base={self.base_layer}, rank={self.rank}, "
+            f"n_adapters={self.n_adapters}, num_experts={self.num_experts}, "
+            f"alpha={self.alpha}, dropout={self.lora_dropout}, "
+            f"use_grouped_mm={self.use_grouped_mm}"
+        )
```
[the remark this answers is not in the document]
Comment (Jackmin801) [reply src/prime_rl/trainer/models/layers/lora/multi_moe.py:437]: ye ok
